internal/api/v1beta1/app_types.go

```diff
@@ -483,6 +487,19 @@ func (s AppStatus) Condition(t ConditionType) *Condition {
 	return nil
 }
 
+func getUpdatedUnits(weight uint8, target uint16) (int, int) {
+	if weight > 100 {
+		weight = 100
+	}
+	// get the split of p1's units and p2's units
+	unitSplit := (float64(weight) / 100) * float64(target)
+	// we want an integer so take the floor of the float and subtract that total from the target
+	p2Units := target - uint16(math.Floor(unitSplit))
```

Comment:
is it possible to use something like `sourceUnits`/`destUnits`/`targetUnits`?
I was a bit confused because in the tests `p1` and `p2` are different processes of an AppDeployment but here they are the same process of two different deployments.

Reply:
Changed.